README
Wx-TreeListCtrl
===============

A simple wrapper around the wxWidgets class wxTreeListCtrl.

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

DEPENDENCIES

This module requires wxWdigets and wxPerl:

COPYRIGHT AND LICENCE

Copyright (C) 2006 by Mark Wardell

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.7 or,
at your option, any later version of Perl 5 you may have available.

This package includes the wxTreeListCtrl source code which you may use
according to the the wxWidgets license.

The authors of the wxTreeListCtrl package are:
	Robert Roebling,
	Julian Smart,
	Alberto Griggio,
	Vadim Zeitlin,
	Otto Wyss,
	Guru Kathiresan



Changes
Revision history for Perl extension Wx::TreeListCtrl.

0.12  Jun  21  2011
    - Used lastest base wxTreeListCtrl code from http://wxcode.sourceforge.net
      several methods updated to take 'coulmn' param.
      Interface should now closely match that described at
      http://wxcode.sourceforge.net/components/treelistctrl/reference.html

0.11  Oct   2  2010
    - Makefile.PL no longer depends on wxperl_overload
    - Better module metadata (Alexandr Ciornii)

0.10  Apr  23  2010

    - Wx::build::MakeMaker added to configure_requires (Alexandr Ciornii)
    - Editable column in example (Alexandr Ciornii)
    - Demo module improvements (Alexandr Ciornii)

0.09  Mar  16  2010
    - Makefile and use fixes
    - version released for PPM repo update.

0.08  Feb  10  2010
    - namespace fix
    
0.07  Feb  10  2010
    - Fixed possible Wx::TreeListColumnInfo thread memory leak
    - Added some tests.   

0.06  Feb   2  2010
    - Updated to use latest source from wxTreeListCtrl
      in wxPython dist but added wxTR_SHOW_ROOT_LABEL_ONLY
      back in.
    - added GetColumn SetColumn GetCurrentItem SetCurrentItem
    - confirmed compilation on wxMAC (carbon), wxMSW, wxGTK (linux)

0.051 A patched non official version.
      get the official version from search.cpan.org

0.05  Mon Aug  21 2006
    - changed build system in accordance with Wx 0.57
    - removed script/pipe.pl

0.04  Fri Aug  18 2006
    - added lib/Wx/DemoModules/wxTreeList.pm

0.03  Thu Aug  17 2006
    - added EXPORT_TAG 'treelist'

0.02  Thu Aug  17 2006
    - fixed exporting of constants

0.01  Tue Aug  8 10:28:54 2006
    - first attempt at wrapping wxTreeListCtrl



lib/Wx/TreeListCtrl.pm
#############################################################################
## Name:        TreeListCtrl.pm
## Purpose:     Wx::TreeListControl
## Author:      Mark Wardell
## Modified by: Mark Dootson
## RCS-ID:      $Id: TreeListCtrl.pm 17 2011-06-21 14:21:11Z mark.dootson $
## Copyright:   (c) 2006-2010 Mark Wardell
## Licence:     This program is free software; you can redistribute it and/or
##              modify it under the same terms as Perl itself
#############################################################################

package Wx::TreeListCtrl;
use 5.008;
use strict;
use warnings;
use Wx;

our $VERSION = '0.12';

our @constants = qw(
     wxTL_MODE_NAV_FULLTREE
     wxTL_MODE_NAV_EXPANDED
     wxTL_MODE_NAV_VISIBLE
     wxTL_MODE_NAV_LEVEL
     wxTL_MODE_FIND_EXACT
     wxTL_MODE_FIND_PARTIAL 
     wxTL_MODE_FIND_NOCASE
     wxTR_HAS_BUTTONS
     wxTR_NO_LINES
     wxTR_LINES_AT_ROOT
     wxTR_TWIST_BUTTONS
     wxTR_MULTIPLE
     wxTR_EXTENDED
     wxTR_HAS_VARIABLE_ROW_HEIGHT
     wxTR_EDIT_LABELS
     wxTR_ROW_LINES
     wxTR_HIDE_ROOT
     wxTR_FULL_ROW_HIGHLIGHT
     wxTR_DEFAULT_STYLE
     wxTR_SINGLE
     wxTR_NO_BUTTONS
     wxTR_VIRTUAL
     wxTR_COLUMN_LINES
     wxTREE_HITTEST_ONITEMCOLUMN
     wxTR_SHOW_ROOT_LABEL_ONLY
);

push @Wx::EXPORT_OK, @constants;
$Wx::EXPORT_TAGS{'treelist'} = [ @constants ];
require XSLoader;
XSLoader::load( 'Wx::TreeListCtrl', $VERSION );

#
# confirm inheritance tree
#

no strict;

package Wx::TreeListCtrl;  @ISA = qw( Wx::Control );

# Alias
*Wx::TreeListCtrl::Edit = \&Wx::TreeListCtrl::EditLabel;

1;
__END__

=head1 NAME

Wx::TreeListCtrl - interface to the Wx::TreeListCtrl class

=head1 VERSION 0.11

=head1 SYNOPSIS

  use Wx::TreeListCtrl;

=head1 DESCRIPTION

Wx::TreeListCtrl is a wrapper for the wxTreeListCtrl class in the wxWidgets GUI toolkit.

=head2 EXPORT

None by default.

=head2 Exportable constants

  wxTL_MODE_NAV_FULLTREE
  wxTL_MODE_NAV_EXPANDED
  wxTL_MODE_NAV_VISIBLE
  wxTL_MODE_NAV_LEVEL
  wxTL_MODE_FIND_EXACT
  wxTL_MODE_FIND_PARTIAL 
  wxTL_MODE_FIND_NOCASE
  wxTR_HAS_BUTTONS
  wxTR_NO_LINES
  wxTR_LINES_AT_ROOT
  wxTR_TWIST_BUTTONS
  wxTR_MULTIPLE
  wxTR_EXTENDED
  wxTR_HAS_VARIABLE_ROW_HEIGHT
  wxTR_EDIT_LABELS
  wxTR_ROW_LINES
  wxTR_HIDE_ROOT
  wxTR_FULL_ROW_HIGHLIGHT
  wxTR_DEFAULT_STYLE
  wxTR_SINGLE
  wxTR_NO_BUTTONS
  wxTR_VIRTUAL
  wxTR_COLUMN_LINES
  wxTREE_HITTEST_ONITEMCOLUMN
  wxTR_SHOW_ROOT_LABEL_ONLY  

=head1 DOCUMENTATION

  The main Wx::TreeListCtrl has the same interface as the wxPython module for
  wxTreeListCtrl which is available here:

  L<http://wxcode.sourceforge.net/components/treelistctrl/reference.html>

  To make column editable (inline) use $control->SetColumnEditable($column_number_from_0,1);

  Wx::TreeListColumnInfo is also available

  my $info = Wx::TreeListColumnInfo->new( coltext, width, flags, imageindex, shown, editable);

     only coltext is required, defaults for other items:
     width      = 100
     flags      = wxALIGN_LEFT
     imageindex = -1
     shown      = 1 (true)
     editable   = 0 (false)

  usage:

  $treelist->AddColumn($info);
  $treelist->InsertColumn(2, $info);
  $treelist->SetColumn(3, $info);

  # set methods have equivalent get methods
  my $info = $treelist->GetColumn(4); 
  $info->SetText('Column Four');
  $info->SetWidth('50');
  $info->SetAlignment(wxALIGN_RIGHT);
  $info->SetImage(1);
  $info->SetSelectedImage(2);
  $info->SetShown(1);
  $info->SetEditable(0);
  $treelist->SetColumn(4, $info);
  $info->SetText('Column Five');
  $treelist->SetColumn(5, $info);

  # note - changing a TreeListColumnInfo object will not
  # affect the underlying object. You must 'SetColumn' to
  # change the underlying data.

  # the two boolean members use 'Is' as a get method
  $info->IsShown
  $info->IsEditable

  The control processes some mouse events internally. To respond to 'right click' and
  context events, use
  EVT_TREE_ITEM_RIGHT_CLICK  - The user has clicked an item with the right mouse button
  EVT_RIGHT_UP - The user has right clicked the control with the right mouse button - but
  not on any item
  EVT_CONTEXT_MENU - The user has used keyboard method to invoke a context menu

=head1 AUTHOR

Mark Wardell <mwardell@cpan.org>

=head1 Current Maintainer

Mark Dootson <mdootson@cpan.org>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2006 - 2011 by Mark Wardell

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.7 or,
at your option, any later version of Perl 5 you may have available.

This package includes the wxTreeListCtrl source code which you may use
according to the the wxWidgets license.

The authors of the wxTreeListCtrl package are:
    Robert Roebling,
    Julian Smart,
    Alberto Griggio,
    Vadim Zeitlin,
    Otto Wyss,
    Guru Kathiresan

=cut
